.circleci/config.yml

```diff
@@ -32,21 +32,8 @@ jobs:
       - run:
           name: Print parameters
           command: cat /tmp/pipeline-parameters.json
-      - run:
-          name: Generate config
-          command: |
-            export INT_MATRIX=$(jq -c 'if (.all) then
-              (["workflow"])
-              else (to_entries | map(select((.value) and (.key | contains("example") | not)) | .key)) end' \
-              /tmp/pipeline-parameters.json)
-            yq '(.workflows.ci.jobs[] | select(."integration-test".matrix.parameters."pytest-group")
-              ."integration-test".matrix.parameters."pytest-group")=env(INT_MATRIX)' \
-              .circleci/continue_config.yml > .circleci/generated_config.yml
-      - run:
-          name: Print generated config
-          command: cat .circleci/generated_config.yml
       - continuation/continue:
-          configuration_path: .circleci/generated_config.yml
+          configuration_path: .circleci/continue_config.yml
```

Comment:
the generated_config.yml is the curple for dataset integration double trigger.

for instance in this pr, here is the generated yaml

```yml
version: 2.1
orbs:
  codecov: codecov/codecov@3.2.4
parameters:
  workflow:
    type: boolean
    default: false
  dataset:
    type: boolean
    default: false
  misc:
    type: boolean
    default: false
  all:
    type: boolean
    default: false
  example:
    type: boolean
    default: false
  token_count:
    type: integer
    default: 8
executors:
  python:
    parameters:
      python-version:
        type: string
        default: "3.9.18"
      resource-class:
        type: string
        default: "small"
    docker:
      - image: cimg/python:<< parameters.python-version >>
    resource_class: << parameters.resource-class >>
    environment:
      POETRY_CACHE_DIR: /home/circleci/project/.poetry
jobs:
  ci-base:
    parameters:
      python-version:
        type: string
        default: "3.9.18"
      extras:
        type: string
        default: "none"
    executor:
      name: python
      python-version: << parameters.python-version >>
    steps:
      - checkout
      - restore_cache:
          key: &ci-base-cache ci-cache-<< parameters.python-version >>-<< parameters.extras >>-{{ checksum "pyproject.toml" }}
      - when:
          condition:
            equal: ["none", << parameters.extras >>]
          steps:
            - run:
                name: Install Dependencies
                command: |
                  poetry config installer.max-workers 10
                  poetry install -vv --no-ansi
      - when:
          condition:
            equal: ["metrics", << parameters.extras >>]
          steps:
            - run:
                name: Install Dependencies
                command: |
                  poetry config installer.max-workers 10
                  poetry install -vv --all-extras --no-ansi
      - save_cache:
          key: *ci-base-cache
          paths:
            - /home/circleci/project/.poetry/virtualenvs
            - poetry.lock
  docs:
    parameters:
      python-version:
        type: string
        default: "3.9.18"
      extras:
        type: string
        default: "none"
    executor:
      name: python
      python-version: << parameters.python-version >>
    steps:
      - checkout
      - restore_cache:
          name: Restore Poetry cache
          key: ci-cache-<< parameters.python-version >>-<< parameters.extras >>-{{ checksum "pyproject.toml" }}
      - restore_cache:
          name: Restore MkDocs cache
          key: mkdocs-cache
      - run:
          name: Install system-level dependencies
          command: |
            sudo apt-get update
            sudo apt-get install -y libcairo2-dev git-lfs
      - run:
          name: Checkout LFS blobs
          command: git lfs pull
      - run:
          name: Build documentation without 'insiders' packages
          command: poetry run mkdocs build
      - add_ssh_keys: # add github.com SSH key fingerprint to access private repo forks
          fingerprints: ["15:8b:d4:ac:1f:cd:2f:d3:92:a7:4c:aa:46:81:0b:7d"]
      - run:
          name: Build documentation with 'insiders' packages
          command: |
            ./docs/setup_insiders.sh
            poetry run mkdocs build --config-file mkdocs.insiders.yml
      - save_cache:
          key: mkdocs-cache
          paths:
            - .cache
  unit-test:
    parameters:
      python-version:
        type: string
        default: "3.9.18"
      extras:
        type: string
        default: "none"
    executor:
      name: python
      python-version: << parameters.python-version >>
    steps:
      - checkout
      - restore_cache:
          key: ci-cache-<< parameters.python-version >>-<< parameters.extras >>-{{ checksum "pyproject.toml" }}
      - run: poetry run python3 -c 'import kolena'
      # TODO: fix underlying mypy issues with Python>3.9 rather than skipping
      - when:
          condition:
            not:
              matches:
                pattern: "^3\\.(10|11).*$"
                value: << parameters.python-version >>
          steps:
            - run: poetry run pre-commit run -a
      - when:
          condition:
            equal: ["none", << parameters.extras >>]
          steps:
            - run:
                name: Run unit tests
                command: poetry run pytest --cache-clear -m 'not metrics' -vv --cov=kolena --cov-branch tests/unit
      - when:
          condition:
            equal: ["metrics", << parameters.extras >>]
          steps:
            - run:
                name: Run metrics unit tests
                command: poetry run pytest --cache-clear -m 'metrics' -vv --cov=kolena --cov-branch tests/unit
      - when:
          # Generate coverage only from one Python version
          condition:
            matches:
              pattern: "^3\\.9.*$"
              value: << parameters.python-version >>
          steps:
            - run:
                name: Coverage
                command: poetry run coverage xml --data-file .coverage
            - codecov/upload:
                file: coverage.xml
                flags: integration
  integration-test:
    parameters:
      python-version:
        type: string
        default: "3.9"
      pytest-group:
        type: string
        default: workflow
      enabled:
        type: boolean
        default: false
      parallelism:
        type: integer
        default: 1
      extras:
        type: string
        default: "none"
    executor:
      name: python
      python-version: << parameters.python-version >>
    parallelism: << parameters.parallelism >>
    steps:
      - when:
          condition:
            not: << parameters.enabled >>
          steps:
            - run:
                name: skip job
                command: circleci-agent step halt
      - checkout
      - restore_cache:
          key: ci-cache-<< parameters.python-version >>-<< parameters.extras >>-{{ checksum "pyproject.toml" }}
      - run:
          name: Set KOLENA_TOKEN with round robin
          command: |
            token=KOLENA_TOKEN_$((CIRCLE_BUILD_NUM % << pipeline.parameters.token_count >>))
            echo "Using $token"
            export KOLENA_TOKEN=${!token}
      - when:
          condition:
            and:
              - equal:
                  - "none"
                  - << parameters.extras >>
              - not_equal:
                  - _experimental
                  - << parameters.pytest-group >>
          steps:
            - run:
                name: Run << parameters.pytest-group >> integration tests
                no_output_timeout: 30m
                command: |
                  TEST_GROUP="<< parameters.pytest-group >>"
                  if [ "$TEST_GROUP" = "misc" ]; then
                    TESTFILES=$(circleci tests glob tests/integration/test_*.py |
                      circleci tests split --split-by=timings --timings-type=filename)
                  else
                    TESTFILES=$(find tests/integration/$TEST_GROUP -name "test_*.py" |
                      circleci tests split --split-by=timings --timings-type=filename)
                  fi
                  poetry run pytest --cache-clear -vv --durations=10 --cov=kolena --cov-branch --ignore=examples \
                    -o junit_family=legacy --junitxml=test-results/result.xml $TESTFILES
      - when:
          condition:
            equal: ["metrics", << parameters.extras >>]
          steps:
            - run:
                name: Run experimental metrics integration tests
                command: poetry run pytest --cache-clear -m 'metrics' -vv --cov=kolena --cov-branch tests/integration/
      - when:
          # Generate coverage only from one python version
          condition:
            matches:
              pattern: "^3\\.9.*$"
              value: << parameters.python-version >>
          steps:
            - run:
                name: Coverage
                command: poetry run coverage xml --data-file .coverage
            - codecov/upload:
                file: coverage.xml
            - store_test_results:
                path: test-results
  example-test-dataset:
    parameters:
      python-version:
        type: string
        default: "3.9.18"
      subproject:
        type: string
      resource-class:
        type: string
    executor:
      name: python
      python-version: << parameters.python-version >>
      resource-class: << parameters.resource-class >>
    working_directory: ~/project/examples/dataset/<< parameters.subproject >>
    steps:
      - checkout:
          path: ~/project
      - run: |
          poetry config installer.max-workers 10
          poetry install -vv --no-ansi
      - run:
          name: Run pre-commit checks
          command: poetry run pre-commit run -a
      - run:
          name: Run << parameters.subproject >> (Python << parameters.python-version >>) integration tests
          command: |
            token=KOLENA_TOKEN_$((CIRCLE_BUILD_NUM % << pipeline.parameters.token_count >>))
            echo "Using $token"
            export KOLENA_TOKEN=${!token}
            poetry run pytest -vv tests
  example-test-workflow:
    parameters:
      python-version:
        type: string
        default: "3.9.18"
      subproject:
        type: string
      resource-class:
        type: string
    executor:
      name: python
      python-version: << parameters.python-version >>
      resource-class: << parameters.resource-class >>
    working_directory: ~/project/examples/workflow/<< parameters.subproject >>
    steps:
      - checkout:
          path: ~/project
      - run: |
          poetry config installer.max-workers 10
          poetry install -vv --no-ansi
      - run:
          name: Run pre-commit checks
          command: poetry run pre-commit run -a
      - run:
          name: Run << parameters.subproject >> (Python << parameters.python-version >>) integration tests
          command: |
            token=KOLENA_TOKEN_$((CIRCLE_BUILD_NUM % << pipeline.parameters.token_count >>))
            echo "Using $token"
            export KOLENA_TOKEN=${!token}
            poetry run pytest --cache-clear -vv tests
  example-evaluator:
    docker:
      - image: cimg/base:2022.06
    steps:
      - checkout:
          path: ~/project
      - setup_remote_docker:
          version: 20.10.14
      - run:
          name: Build evaluator
          command: examples/workflow/text_summarization/docker/build.sh
workflows:
  ci:
    jobs:
      - ci-base:
          name: ci-base-<< matrix.python-version >>-<< matrix.extras >>
          matrix:
            parameters:
              python-version: ["3.8.18", "3.9.18", "3.10.13", "3.11.6"]
              extras: ["none", "metrics"]
      - unit-test:
          name: unit-test-<< matrix.python-version >>-<< matrix.extras >>
          matrix:
            parameters:
              python-version: ["3.8.18", "3.9.18", "3.10.13", "3.11.6"]
              extras: ["none", "metrics"]
          requires:
            - ci-base-<< matrix.python-version >>-<< matrix.extras >>
      - docs:
          name: docs-<< matrix.python-version >>
          matrix:
            parameters:
              python-version: ["3.9.18"]
              extras: ["none"]
          requires:
            - ci-base-<< matrix.python-version >>-<< matrix.extras >>
      - integration-test:
          name: integration-test-<< matrix.pytest-group >>-<< matrix.python-version >>-<< matrix.extras >>
          matrix:
            parameters:
              python-version: ["3.9.18"]
              pytest-group: ["dataset"]
              extras: ["none"]
          enabled: true
          parallelism: 3
          requires:
            - ci-base-<< matrix.python-version >>-<< matrix.extras >>
      - integration-test:
          name: integration-test-experimental-<< matrix.python-version >>-<< matrix.extras >>
          matrix:
            parameters:
              python-version: ["3.9.18"]
              extras: ["metrics"]
          pytest-group: _experimental
          enabled: true
          requires:
            - ci-base-<< matrix.python-version >>-<< matrix.extras >>
      - integration-test:
          name: integration-test-dataset-<< matrix.python-version >>-<< matrix.extras >>
          matrix:
            parameters:
              python-version: ["3.9.18"]
              extras: ["none"]
          pytest-group: dataset
          enabled: true
          requires:
            - ci-base-<< matrix.python-version >>-<< matrix.extras >>
      - integration-test:
          name: integration-test-misc-<< matrix.python-version >>-<< matrix.extras >>
          matrix:
            parameters:
              python-version: ["3.9.18"]
              extras: ["none"]
          pytest-group: misc
          enabled: << pipeline.parameters.all >>
          parallelism: 1
          requires:
            - ci-base-<< matrix.python-version >>-<< matrix.extras >>
  example:
    when:
      or: [<< pipeline.parameters.workflow >>, << pipeline.parameters.all >>, << pipeline.parameters.example >>]
    jobs:
      - example-test-dataset:
          matrix:
            parameters:
              subproject: [text_summarization]
              resource-class: [large]
              python-version: ["3.9.18"]
      - example-test-dataset:
          matrix:
            parameters:
              subproject: [classification, keypoint_detection, question_answering, rain_forecast, speaker_diarization, age_estimation]
              resource-class: [small]
              python-version: ["3.9.18"]
      - example-test-workflow:
          matrix:
            parameters:
              subproject: [text_summarization]
              resource-class: [large]
              python-version: ["3.9.18"]
      - example-test-workflow:
          matrix:
            parameters:
              subproject: [keypoint_detection, age_estimation, object_detection_3d, classification, question_answering, face_recognition_11, speaker_diarization, automatic_speech_recognition]
              resource-class: [small]
              python-version: ["3.9.18"]
  example-evaluator:
    when:
      condition:
        equal: ["trunk", << pipeline.git.branch >>]
    jobs:
      - example-evaluator
```

which explains why there are two jobs for dataset, one with 3 parallel and another with 1